renderdoc/driver/vulkan/vk_resources.cpp

```diff
@@ -3948,6 +3948,107 @@ void ImgRefs::Split(bool splitAspects, bool splitLevels, bool splitLayers)
   areLayersSplit = newSplitLayerCount > 1;
 }
 
+void QueryPoolInfo::Add(uint32_t firstQuery, rdcarray<uint64_t> values)
+{
+  Reset(firstQuery, (uint32_t)values.size());
+
+  m_Entries.reserve(m_Entries.size() + values.size());
+  for(uint64_t value : values)
+    m_Entries.emplace_back(firstQuery++, value);
+
+  std::sort(m_Entries.begin(), m_Entries.end());
+}
+
+void QueryPoolInfo::Reset(uint32_t firstQuery, uint32_t queryCount)
+{
+  m_Entries.removeIf([&](const auto &entry) {
```

Comment:
Please avoid `auto` except where necessary for lambdas or extremely ugly for std iterators, you can use `Entry` here.

Also please use explicit lambda capture rather than unconditional capture to be clear about what is stored in the lambda.